src/websocket/server.cc

```diff
@@ -123,14 +117,23 @@ future<> connection::process() {
 }
 
 static std::string sha1_base64(std::string_view source) {
-    // CryptoPP insists on freeing the pointers by itself...
-    // It's leaky, but `read_http_upgrade_request` is a one-shot operation
-    // per handshake, so the real risk is not particularly great.
-    CryptoPP::SHA1 sha1;
-    std::string hash;
-    CryptoPP::StringSource(reinterpret_cast<const CryptoPP::byte*>(source.data()), source.size(),
-            true, new CryptoPP::HashFilter(sha1, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(hash), false)));
-    return hash;
+    unsigned char hash[20];
```

Comment:
Nitpick: does GNUTLS have a macro for SHA1 length that perhaps we can use instead of "20"?

Reply:
no, it does not. i checked both GnuTLS and searched around for projects using this library. no luck. i put more details in the commit message as well.

Reply:
I think the size of the hash is unlikely to ever change.